packages/compose/src/hooks/use-global-event/index.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Adds a listener to the containing window.
+ *
+ * @param {Object} ref          A reference with a node that is contained by the
+ *                              window.
+ * @param {Array}  args         `addEventListener` arguments.
+ * @param {Array}  dependencies Hook dependencies.
+ */
+export default function useGlobalEvent( ref, args, dependencies ) {
```

Comment:
Of of the reasons for the existence  of withGlobalEvent is the performance impact of using a single event listener for multiple components using  the same args. Do we want to support that here?

Reply:
I forget where exactly I'd remarked about it, but over time I started to question whether there really should expect to be any performance benefit of using one event handler for all global events, since in either case _something_ would need to iterate and invoke each of the handlers. It seems browser native code could likely be more efficient about this.

Reply:
> I forget where exactly I'd remarked about it

See https://github.com/WordPress/gutenberg/pull/18816#issuecomment-564701641